js/app/application.js

```diff
@@ -245,8 +246,23 @@ const Application = new Knowledge.Class({
         });
     },
 
+    initialize_set_map: function () {
+        Eknc.Engine.get_default().query_promise(Eknc.QueryObject.new_from_props({
+            limit: GLib.MAXUINT32,
+            tags_match_all: ['EknSetObject'],
+        }))
+        .then((results) => {
+            SetMap.init_map_with_models(results.models);
+        })
+        .catch(function (error) {
+            logError(error, 'Failed to load sets from database');
+        });
+    },
+
     _ensure_controller: function () {
         if (this._controller === null) {
+            this.initialize_set_map();
```

Comment:
@ptomato we could the init call right after we setup the default domain, on _check_for_content, but, I don't think that guarantees anything (or at least not as I understand it).